addons/website/static/src/components/fields/fields.js

```diff
@@ -1,24 +1,32 @@
 /** @odoo-module **/
 
 import {PageDependencies} from '@website/components/dialog/page_properties';
+import {standardFieldProps} from '@web/views/fields/standard_field_props';
+import {useInputField} from '@web/views/fields/input_field_hook';
+import {useService} from '@web/core/utils/hooks';
 import {Switch} from '@website/components/switch/switch';
 import AbstractFieldOwl from 'web.AbstractFieldOwl';
 import fieldRegistry from 'web.field_registry_owl';
+import {registry} from '@web/core/registry';
+import {formatChar} from '@web/views/fields/formatters';
 
-const {useState, onWillStart} = owl;
+const {Component, useState, onWillStart} = owl;
 
-class FieldPageUrl extends AbstractFieldOwl {
+/**
+ * Displays website page dependencies and URL redirect options when the page URL
+ * is updated.
+ */
+class PageUrlField extends Component {
```

Comment:
Yes we don't need to use the `AbstractField`. Actually we had a different structure in the old version:

A- Owl specific fields should extend: [`AbstractField` which extends `LegacyComponent`(*)], which is just a translation of `AbstractField` widget features to Owl with an override on `BasicRenderer` to handle both legacy & Owl fields on legcay views (details [here](https://github.com/odoo/odoo/commit/3fd7b2009ec1f16c56c6d0ae87004011c90e204c)).

B- But now since the [new Owl views structure](https://github.com/odoo/odoo/commit/48ef812a635f70571b395f82ffdb2969ce99da9e):

Fields can be added as normal Components with some updates to make the new views API work (e.g. we add related field props (see: `standardFieldProps`), the `extractProps` API used to extract data from the field in view arch instead of having them ready on widget in legacy code...)

I didn't found any documentation for the new structure but I think you can see in /web > `FormController`, `FormRenderer`, `View` and `Field` components how this was implemented :+1: 

(*): LegacyComponent is depercated now.